File: EvoNet/Core/World.cs
namespace EvoNet.Core;

public class World
{
    public int Width { get; init; }
    
    public int Height { get; init; }
    
    public Brain Brain { get; init; }
    
    public WorldAgent[] Agents { get; private set; }

    public void SetAgents(Agent[] agents)
    {
        Agents = new WorldAgent[agents.Length];
        
        for (int i = 0; i < Agents.Length; i++)
        {
            var agent = agents[i];
            
            Agents[i] = new()
            {
                Agent = agent,
                X = Random.Shared.Next(Width),
                Y = Random.Shared.Next(Height)
            };
        }
    }
}

File: EvoNet/World.fs
module EvoNet.World

open EvoNet.Agent

type World = {
    Width: int
    Height: int
    Agents: (Agent * int * int) list
}

let position agent world =
    let _, x, y = world.Agents
                |> List.find (fun (a, _, _) -> a = agent)
    
    x, y
    
let move agent world (x, y) =
    let agents = world.Agents
                 |> List.map (fun t -> match t with
                                       | a, _, _ when a = agent -> (agent, x, y)
                                       | _ -> t)
    
    {
        Agents = agents
        Width = world.Width
        Height = world.Height
    }
